irods_specific_implementation/irods_rest_admin_api_implementation.h

```diff
@@ -73,6 +93,38 @@ namespace irods::rest
                 return make_error_response(SYS_INVALID_INPUT_PARAM, e.what());
             }
         } // operator()
+
+        std::string obfuscate_password(const std::string_view _new_password)
+        {
+            std::array<char, MAX_PASSWORD_LEN + 10> plain_text_password{};
+            std::strncpy(plain_text_password.data(), _new_password.data(), MAX_PASSWORD_LEN);
+
+            if (const auto lcopy = MAX_PASSWORD_LEN - 10 - _new_password.size(); lcopy > 15) {
+                // The random string (second argument) is used for padding and must match 
+                // what is defined on the server-side.
+                std::strncat(plain_text_password.data(), "1gCBizHWbwIYyWLoysGzTe6SyzqFKMniZX05faZHWAwQKXf6Fs", lcopy);
+            }
+
+            std::array<char, MAX_PASSWORD_LEN + 10> admin_password{};
+
+            // Get the plain text password of the iRODS service account user.
+            // "obfGetPw" decodes the obfuscated password stored in .irods/.irodsA.
+            if (obfGetPw(admin_password.data()) != 0) {
+                THROW(SYS_INTERNAL_ERR, "failed to unobfuscate admin password in .irodsA file.");
```

Comment:
consider using a more descriptive error code

Reply:
nothing in the 900000 section seems quite correct.

nothing with 'auth' or 'pass' seems quite right.  

could be fine leaving this here.   nothing the user can do to fix this...

Reply:
sure. i didn't spot anything in the error code table that seemed much better.

leaving as is.